src/reducers/url-state.js

```diff
@@ -119,13 +119,36 @@ const selectedThreads: Reducer<Set<ThreadIndex> | null> = (
 ) => {
   switch (action.type) {
     case 'CHANGE_SELECTED_THREAD':
-    case 'SELECT_TRACK':
+    case 'SELECT_TRACK': {
+      console.log(state, '123');
+      return state;
+    }
     case 'VIEW_FULL_PROFILE':
     case 'VIEW_ORIGINS_PROFILE':
     case 'VIEW_ACTIVE_TAB_PROFILE':
     case 'ISOLATE_PROCESS':
     case 'ISOLATE_PROCESS_MAIN_THREAD':
     case 'HIDE_GLOBAL_TRACK':
+    case 'HIDE_TRACK_BY_TYPE': {
+      // Need some help here!
+      const { globalIdsToHide, localIdsToHide } = action;
+      if (state === null) {
+        // Either there was no selected thread, or the thread indexes were not modified.
+        return state;
+      }
+      const newSelectedThreads = new Set();
+      for (const oldThreadIndex of state) {
+        if (!globalIdsToHide.includes(oldThreadIndex)) {
+          newSelectedThreads.add(oldThreadIndex);
+        }
+        for (const idsToHideforPid of localIdsToHide.values()) {
+          if (!idsToHideforPid.includes(oldThreadIndex)) {
+            newSelectedThreads.add(oldThreadIndex);
+          }
+        }
```

Comment:
Hi @julienw ! 

>In this file, you'll also need to update the reducer selectedThreads: indeed if one of the selected threads is hidden, we want to unselect it. It needs to be computed in the action function. You can look at how we do it for HIDE_GLOBAL_TRACK as an example.

I need some help here! not 100% sure how the `state` looks in this reducer, so I think this code should be incorrect! Could you please help me here? Thanks!

Reply:
In this reducer, the state is a Set of ThreadIndex. Indeed the user can select several tracks with ctrl + click.

In this reducer, for a lot of actions, we did actually the reducer work in the action, so in the reducer we reused `selectedThreadIndexes`. We did that because we don't only need to unselect a thread, we also need to select a new one!

I believe we need to do the same thing here as well, we can't do this action from the reducer only.